src/pybind/mgr/dashboard_v2/controllers/filesystem.py

```diff
@@ -0,0 +1,357 @@
+        # except AttributeError:
+        #     raise cherrypy.HTTPError(404,
+        #                              "No filesystem with id {0}".format(fs_id))
+        if clients is None:
+            raise cherrypy.HTTPError(404,
+                                     "No filesystem with id {0}".format(fs_id))
+        # TODO do something sensible with status
+
+        # Decorate the metadata with some fields that will be
+        # indepdendent of whether it's a kernel or userspace
+        # client, so that the javascript doesn't have to grok that.
+        for client in clients:
+            if "ceph_version" in client['client_metadata']:
+                client['type'] = "userspace"
+                client['version'] = client['client_metadata']['ceph_version']
+                client['hostname'] = client['client_metadata']['hostname']
+            elif "kernel_version" in client['client_metadata']:
+                client['type'] = "kernel"
+                client['version'] = client['client_metadata']['kernel_version']
+                client['hostname'] = client['client_metadata']['hostname']
+            else:
+                client['type'] = "unknown"
+                client['version'] = ""
+                client['hostname'] = ""
+
+        return clients
+
+    def get_latest(self, daemon_type, daemon_name, stat):
+        data = self.mgr.get_counter(daemon_type, daemon_name, stat)[stat]
+        if data:
+            return data[-1][1]
+        else:
+            return 0
+
+    def format_dimless(self, n, width, colored=True):
+        """
+        Format a number without units, so as to fit into `width` characters, substituting
+        an appropriate unit suffix.
+        """
+        units = [' ', 'k', 'M', 'G', 'T', 'P']
+        unit = 0
+        while len("%s" % (int(n) // (1000**unit))) > width - 1:
+            unit += 1
+
+        if unit > 0:
+            truncated_float = ("%f" % (n / (1000.0 ** unit)))[0:width - 1]
+            if truncated_float[-1] == '.':
+                truncated_float = " " + truncated_float[0:-1]
+        else:
+            truncated_float = "%{wid}d".format(wid=width - 1) % n
+        formatted = "%s%s" % (truncated_float, units[unit])
+
+        if colored:
+            # TODO: html equivalent
+            # if n == 0:
+            #     color = self.BLACK, False
+            # else:
+            #     color = self.YELLOW, False
+            # return self.bold(self.colorize(formatted[0:-1], color[0], color[1])) \
+            #     + self.bold(self.colorize(formatted[-1], self.BLACK, False))
+            return formatted
+        else:
+            return formatted
+
+
+class CephFSClients(object):
+    def __init__(self, module_inst, fscid):
+        self._module = module_inst
+        self.fscid = fscid
+
+    def get(self):
+        mds_spec = "{0}:0".format(self.fscid)
+        result = CommandResult("")
+        self._module.send_command(result, "mds", mds_spec,
```

Comment:
If possible I wouldn't directly use self._module.send_command, as it is awkward to use. E.g. we have `models/send_command_api.py`